server/store/sqlstore/group.go

```diff
@@ -0,0 +1,50 @@
+package sqlstore
+
+import (
+	"strings"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+var escapeLikeSearchChar = []string{
+	"%",
+	"_",
+}
+
+func sanitizeSearchTerm(term string) string {
+	const escapeChar = "\\"
+
+	term = strings.ReplaceAll(term, escapeChar, "")
+
+	for _, c := range escapeLikeSearchChar {
+		term = strings.ReplaceAll(term, c, escapeChar+c)
+	}
+
+	return term
+}
+
+func (ss SQLStore) SearchGroupsByPrefix(prefix string) ([]*model.Group, error) {
+	sanitizedPrefix := strings.ToLower(sanitizeSearchTerm(prefix))
+	query := ss.replicaBuilder.
+		Select("Id", "DisplayName", "DeleteAt").
+		From("UserGroups").
+		Where(sq.Like{"LOWER(DisplayName)": sanitizedPrefix + "%"}).
+		Where(sq.Eq{"DeleteAt": 0}).
```

Comment:
Is this meant to be for LDAP groups only or will it also include custom groups?

Reply:
It will work for any group, not just LDAP groups. Do we need to limit it to LDAP groups? I think this feature could be useful for other types, too.

Reply:
I think we limit it to LDAP groups for now. Custom groups are usually loose on permissions so users can join/leave as they please. LDAP groups are managed by an admin